### PR TITLE
feat: Add `split,MsBackend` and `split,Spectra`

### PR DESCRIPTION
**Pull request by jorainer**
- Add `split,MsBackend`, `split,MsBackendDataFrame` (issue #64)
- Add `split,Spectra` (issue #66)

Performance improvement due to `split,MsBackendDataFrame`: https://github.com/rformassspectrometry/Spectra/issues/64#issuecomment-531656701

This will make `lapply,Spectra` faster.

### CODE REVIEW

**Comment by jorainer** [comment]
Sorry for the latest commit - I've just fixed also the performance issues mentioned in issue #68, mostly by removing code that is no longer needed since we're using `NumericList` to store `mz` and `intensity` values.

File: R/MsBackendMzR-functions.R [hunk 1]
```diff
@@ -37,6 +37,7 @@ MsBackendMzR <- function() {
         hdr$isolationWindowUpperOffset
     hdr$isolationWindowUpperOffset <- NULL
     hdr$isolationWindowLowerOffset <- NULL
+    ## Remove core spectra variables that contain only `NA`
```
Comment (sgibb) [inline R/MsBackendMzR-functions.R]: You want to do this in the future, or (#67)?

File: R/MsBackendDataFrame.R [hunk 2]
```diff
@@ -521,6 +521,16 @@ setMethod("[", "MsBackendDataFrame", function(x, i, j, ..., drop = FALSE) {
     .subset_backend_data_frame(x, i)
 })
 
+#' @rdname hidden_aliases
+setMethod("split", "MsBackendDataFrame", function(x, f, drop = FALSE, ...) {
+    if (!is.factor(f))
+        f <- as.factor(f)
+    if (length(levels(f)) > (nrow(x@spectraData) / 10))
```
[the remark this answers is not in the document]
Comment (jorainer) [reply R/MsBackendDataFrame.R]: Yep, it is empiric. It was based on a *normal* spectra data *DFrame* from an mzML file, many `Rle` being only `NA`.

And yes, it gets slower with more levels (that translate into the loop). It's very slow if you split by individual rows, but it is fast if you split the `DFrame` by a factor with 2 levels. It's always the tradeoff between the time it takes to convert the `Rle` in the `DFrame` into basic data types and the time it takes to split the `DFrame`.

Note also that splitting a `data.frame` would be way faster.

File: R/MsBackendMzR-functions.R (hunk 1 above)
Comment (jorainer) [reply R/MsBackendMzR-functions.R]: I have it already in a local branch - just didn't want to push that to this pull request to not overcomplicate it (and to not push additional features to an already created PR).

File: R/MsBackendDataFrame.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (sgibb) [reply R/MsBackendDataFrame.R]: So maybe we need to rediscuss the decision to use `DFrame` and/or `Rle`?

[the remark this answers is not in the document]
Comment (jorainer) [reply R/MsBackendDataFrame.R]: In the longer run yes - or we could even think of implementing a `MsBackenddata.frame` (lovely name, isn't it?) to see how/if there are major performance issues.

**Pull request merged**